app/storage/postgres/tenant.go

```diff
@@ -214,6 +222,12 @@ func (s *TenantStorage) UpdateSettings(settings *models.UpdateTenantSettings) er
 
 		// Update reference to new logo ID
 		s.current.LogoID = int(newLogoID.Int64)
+
+		if oldLogoExists {
+			if _, err = s.trx.Execute(`DELETE FROM uploads WHERE id = $1`, oldLogoID); err != nil {
```

Comment:
```suggestion
			if _, err = s.trx.Execute(`DELETE FROM uploads WHERE id = $1 AND tenant_id = $2`, oldLogoID, s.current.ID); err != nil {
```

Same as above, we always filter by tenant_id.